### PR TITLE
[.NET] Korean NumberWithUnit partial support

### PR DESCRIPTION
**Pull request by LionbridgeCSII**
AgeModel: 8 pass, 1 only extraction, 7 fail.
CurrencyModel: 50 pass, 16 only extraction, 42 fail.
DimensionModel: 41 pass, 11 fail.
TemperatureModel: 8 pass, 24 only extraction.

The not resolved and not supported cases are due mostly to problems in the Number Recognizer which is not fully developed and fails to extract or parse certain numbers (especially large ones). Other issues are given by the Korean way of handling units (e.g. splitting them around the number like in "시속 188 마일" literally "per hour 188 miles").

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Text.NumberWithUnit/NumberWithUnitRecognizer.cs [hunk 1]
```diff
@@ -551,6 +551,62 @@ protected override void InitializeConfiguration()
                     },
                 }));
 
+            RegisterModel<CurrencyModel>(
+                Culture.Korean,
+                (options) => new CurrencyModel(new Dictionary<IExtractor, IParser>
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.NumberWithUnit/NumberWithUnitRecognizer.cs:556]: Minor: Please follow alphabetic order per culture.

File: Patterns/Korean/Korean-NumbersWithUnit.yaml [hunk 2]
```diff
@@ -0,0 +1,781 @@
...
+  entries:
+    - 살|년|세|일|주|월|개월
+AgeSuffixList: !dictionary
+  types: [string, string]
+  entries:
+    Year: 살|년|세
+    Month: 월|개월
+    Week: 주
+    Day: 일
+#ChineseNumberWithUnitExtractorConfiguration
```
Comment (tellarin) [inline Patterns/Korean/Korean-NumbersWithUnit.yaml:14]: Minor: Update comment.

File: Patterns/Korean/Korean-NumbersWithUnit.yaml [hunk 3]
```diff
@@ -0,0 +1,781 @@
...
+    Dominican peso: 도미니칸 페소 #多米尼加比索
+    Mexican peso: 멕시코 페소 #墨西哥比索
+    Philippine peso: 필리핀 페소 #菲律宾比索
+    Uruguayan peso: 우루과이 페소 #乌拉圭比索
+    Peso: 페소 #比索
+    Centavo: 仙|菲辅币
+#Pound
+    Alderney pound: 奥尔德尼镑
+    British pound: 영국 파운드 #英镑
+    Guernsey pound: 건지섬 파운드 #https://ko.wikipedia.org/wiki/%EA%B1%B4%EC%A7%80%EC%84%AC_%ED%8C%8C%EC%9A%B4%EB%93%9C #根西镑
```
Comment (tellarin) [inline Patterns/Korean/Korean-NumbersWithUnit.yaml:267]: What do these commented out cases at the end mean?

File: Patterns/Korean/Korean-NumbersWithUnit.yaml [hunk 4]
```diff
@@ -0,0 +1,781 @@
...
+    - 란드
+    - 엔
+    - 센트
+    - 프랑
+    - 휠던
+#DimensionExtractorConfiguration
+DimensionSuffixList: !dictionary
+  types: [string, string]
+  entries:
+# China 公制
```
Comment (tellarin) [inline Patterns/Korean/Korean-NumbersWithUnit.yaml:623]: ???

File: Patterns/Korean/Korean-NumbersWithUnit.yaml [hunk 5]
```diff
@@ -0,0 +1,781 @@
...
+    Jersey pound: 저지섬 파운드 #https://ko.wikipedia.org/wiki/%EC%A0%80%EC%A7%80%EC%84%AC_%ED%8C%8C%EC%9A%B4%EB%93%9C #泽西岛镑
+    Lebanese pound: 레바논 파운드 #黎巴嫩镑
+    South Sudanese pound: 남수단 파운드 #南苏丹镑
+    Sudanese pound: 수단 파운드 #苏丹镑
+    Syrian pound: 시리아 파운드 #叙利亚镑
+    Pound: 파운드
+    Pence: 펜스
+    Shilling: 先令
+#Dollar
+    United States dollar: US 달러|미국 달러 #美元|美金|美圆
```
Comment (tellarin) [inline Patterns/Korean/Korean-NumbersWithUnit.yaml:282]: In no informal case people use foreign words for currencies or other units?
If informally they show up in text or speech, we should cover them.

File: Specs/NumberWithUnit/Korean/AgeModel.json [hunk 6]
```diff
@@ -1,7 +1,8 @@
 [
   {
     "Input": "그녀가 다섯 살 때, 그녀는 자전거 타는 것을 배웠다.",
-    "NotSupportedByDesign": "dotNet,javascript,python,java,",
+    "NotSupportedByDesign": "javascript,python,java",
+    "IgnoreResolution": "true",
```
Comment (tellarin) [inline Specs/NumberWithUnit/Korean/AgeModel.json:5]: For such cases, can you add a "Comment" attribute with the reason? For example, if it's a number failure or some other case?

File: Specs/NumberWithUnit/Korean/AgeModel.json [hunk 7]
```diff
@@ -57,7 +59,8 @@
   },
   {
     "Input": "중국의 만리장성은 500년 이상 되었고, 5,000마일 이상 뻗어 있다.",
-    "NotSupportedByDesign": "dotNet,javascript,python,java,",
+    "NotSupportedByDesign": "javascript,python,java",
+    "NotSupported": "dotnet",
```
Comment (tellarin) [inline Specs/NumberWithUnit/Korean/AgeModel.json:63]: Same.

**Review by tellarin — CHANGES_REQUESTED** [review]

File: .NET/Microsoft.Recognizers.Text.NumberWithUnit/NumberWithUnitRecognizer.cs (hunk 1 above)
Comment (LionbridgeCSII) [reply .NET/Microsoft.Recognizers.Text.NumberWithUnit/NumberWithUnitRecognizer.cs:556]: reordered models

File: Patterns/Korean/Korean-NumbersWithUnit.yaml (hunk 2 above)
Comment (LionbridgeCSII) [reply Patterns/Korean/Korean-NumbersWithUnit.yaml:14]: done

File: Patterns/Korean/Korean-NumbersWithUnit.yaml (hunk 3 above)
Comment (LionbridgeCSII) [reply Patterns/Korean/Korean-NumbersWithUnit.yaml:267]: removed

File: Patterns/Korean/Korean-NumbersWithUnit.yaml (hunk 4 above)
Comment (LionbridgeCSII) [reply Patterns/Korean/Korean-NumbersWithUnit.yaml:623]: removed

File: Patterns/Korean/Korean-NumbersWithUnit.yaml (hunk 5 above)
Comment (LionbridgeCSII) [reply Patterns/Korean/Korean-NumbersWithUnit.yaml:282]: it seems they are not really used. Symbols ($,£,€...) are used but only as prefixes not as suffixes.

File: Specs/NumberWithUnit/Korean/AgeModel.json (hunk 6 above)
Comment (LionbridgeCSII) [reply Specs/NumberWithUnit/Korean/AgeModel.json:5]: comments added.

File: Specs/NumberWithUnit/Korean/AgeModel.json (hunk 7 above)
Comment (LionbridgeCSII) [reply Specs/NumberWithUnit/Korean/AgeModel.json:63]: added.

**Comment by LionbridgeCSII** [comment]
After reviewing AgeModel more carefully, we decided to keep only the unambiguous age units to avoids conflicts with duration and date, and we modified a couple of inputs to take this into account:
"십 년" (ten years) -> "십 세" (ten years old)
"500년" (500 years) -> "500살" (500 years old)

**Comment by tellarin** [comment]
> After reviewing AgeModel more carefully, we decided to keep only the unambiguous age units to avoid conflicts with duration and date, and we modified a couple of inputs to take this into account:
> "십 년" (ten years) -> "십 세" (ten years old)
> "500년" (500 years) -> "500살" (500 years old)

In this case, perhaps it would be good to keep the ambiguous specs and leave them marked as NotSupportedByDesign with a Comment for the reason. Having such cases in the spec set will be very helpful in future iterations of the project.

**Comment by LionbridgeCSII** [comment]
> In this case, perhaps it would be good to keep the ambiguous specs and leave them marked as NotSupportedByDesign with a Comment for the reason. Having such cases in the spec set will be very helpful in future iterations of the project.

Ok, reintroduced with comment.

**Review by tellarin — APPROVED** [review]

**Pull request merged**